### PR TITLE
Validate intents for GatewayDiscordClient::requestMembers

### PR DESCRIPTION
**Pull request by TheNumberOne**
**Description:** Adds validation for the `requestMembers()` function.

**Justification:** For https://github.com/Discord4J/Discord4J/issues/684

### CODE REVIEW

File: core/src/main/java/discord4j/core/util/ValidationUtil.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     * @param possibleIntents The current intents
+     * @see <a href="https://discord.com/developers/docs/topics/gateway#request-guild-members">https://discord.com/developers/docs/topics/gateway#request-guild-members</a>
+     */
+    public static void validateRequestGuildMembers(RequestGuildMembers request, Possible<IntentSet> possibleIntents) {
+        if (request.query().isAbsent() == request.userIds().isAbsent()) {
+            throw new IllegalArgumentException("One of query or user ids is required.");
+        }
+
+        // Further Validation is only required if we are using intents.
+        if (possibleIntents.isAbsent()) return;
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/util/ValidationUtil.java:22]: Please use braces around this return statement

File: core/src/test/java/discord4j/core/util/ValidationUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package discord4j.core.util;
```
Comment (quanticc) [inline core/src/test/java/discord4j/core/util/ValidationUtilTest.java:1]: Missing license header

File: core/src/test/java/discord4j/core/ExampleIntents.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package discord4j.core;
```
Comment (quanticc) [inline core/src/test/java/discord4j/core/ExampleIntents.java:1]: Missing license header

File: core/src/main/java/discord4j/core/util/ValidationUtil.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package discord4j.core.util;
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/util/ValidationUtil.java:1]: Missing license header

**Comment by TheNumberOne** [comment]
Coolio. I'll get those fixes in and rebase

**Comment by TheNumberOne** [comment]
Everything should be fixed as requested! Also, for pull requests, do y'all prefer peeps rebasing their changes onto master? Or continuously merging master into the pull request? For this pull request, I did the rebasing style. But, if you all prefer the merge style I can do that for future pull requests.

**Comment by Shadorc** [comment]
The risk with this is to throw exceptions on start-up if a bot does not have GUILD_MEMBERS intent and have the default MemberRequestFilter (requesting full member list on large guilds). It could be avoided by adjusting the MemberRequestFilter depending on the intent or just disabling it by default, but it's probably out of scope for this PR.

**Comment by quanticc** [comment]
@Shadorc I don't think I can reproduce that scenario, can you? You can use this code to try the PR:
```
git checkout -b TheNumberOne-master master
git pull git://github.com/TheNumberOne/Discord4J.git master
```

@TheNumberOne we squash commits when merging to our branch, unless in cases where we want to preserve the history for semantic reasons, particularly for large PRs

**Pull request merged**